frontend-api/src/main/java/uk/gov/di/authentication/frontendapi/lambda/MfaResetAuthorizeHandler.java

```diff
@@ -4,14 +4,117 @@
+        super(
+                MfaResetRequest.class,
+                configurationService,
+                sessionService,
+                clientSessionService,
+                clientService,
+                authenticationService);
+        this.mfaResetIPVAuthorizationService = mfaResetIPVAuthorizationService;
+    }
+
+    public MfaResetAuthorizeHandler(ConfigurationService configurationService) {
+        super(MfaResetRequest.class, configurationService);
+        RedisConnectionService redisConnectionService =
+                new RedisConnectionService(configurationService);
+        KmsConnectionService kmsConnectionService = new KmsConnectionService(configurationService);
+        JwtService jwtService = new JwtService(kmsConnectionService);
+        TokenService tokenService =
+                new TokenService(
+                        configurationService, redisConnectionService, kmsConnectionService);
+        this.mfaResetIPVAuthorizationService =
+                new MfaResetIPVAuthorizationService(
+                        configurationService,
+                        jwtService,
+                        tokenService,
+                        redisConnectionService,
+                        new AuditService(configurationService),
+                        new CloudwatchMetricsService(configurationService));
+    }
+
+    public MfaResetAuthorizeHandler() {
+        this(ConfigurationService.getInstance());
+    }
+
     @Override
     public APIGatewayProxyResponseEvent handleRequest(
             APIGatewayProxyRequestEvent input, Context context) {
-        return generateApiGatewayProxyResponse(200, "hello world");
+        return super.handleRequest(input, context);
+    }
+
+    @Override
+    public APIGatewayProxyResponseEvent handleRequestWithUserContext(
+            APIGatewayProxyRequestEvent input,
+            Context context,
+            MfaResetRequest request,
+            UserContext userContext) {
+        LOG.info("MFA Reset Authorization request received");
+        try {
+            Session userSession = userContext.getSession();
+            String clientSessionId = userContext.getClientSessionId();
+            attachSessionIdToLogs(userSession.getSessionId());
+
+            AuditContext auditContext =
+                    new AuditContext(
+                            userContext.getClientId(),
+                            userContext.getClientSessionId(),
+                            userContext.getSession().getSessionId(),
+                            AuditService.UNKNOWN,
+                            request.getEmail(),
+                            IpAddressHelper.extractIpAddress(input),
+                            AuditService.UNKNOWN,
+                            PersistentIdHelper.extractPersistentIdFromHeaders(input.getHeaders()),
+                            Optional.ofNullable(userContext.getTxmaAuditEncoded()));
+
+            Subject internalCommonSubjectId =
```

Comment:
We should start documenting the internal interface for this, and put this assumption there: that this value will always be set by the time this function is called.

Reply:
Have started to document this interface [here](https://govukverify.atlassian.net/wiki/spaces/LO/pages/4341661806/Backend+User+Session+Store+interface), currently a bit WIP but happy for any feedback.